relay-server/src/actors/outcome.rs

```diff
@@ -835,17 +820,46 @@ impl OutcomeProducerService {
         let key = message.event_id.unwrap_or_else(EventId::new).0;
 
         // Dispatch to the correct topic and cluster based on the kind of outcome.
-        let (topic, producer) = if message.is_billing() {
+        let (_topic, producer) = if message.is_billing() {
             (KafkaTopic::OutcomesBilling, producer.billing())
         } else {
             (KafkaTopic::Outcomes, producer.default())
         };
 
-        let record = BaseRecord::to(self.config.kafka_topic_name(topic))
-            .payload(&payload)
-            .key(key.as_bytes().as_ref());
+        let result = match producer {
+            Producer::Single {
+                topic_name,
+                producer,
+            } => {
+                let record = BaseRecord::to(topic_name)
+                    .payload(&payload)
+                    .key(key.as_bytes().as_ref());
+
+                producer.send(record)
+            }
+
+            Producer::Sharded { shards, producers } => {
+                let mut hasher = FnvHasher::default();
+                std::hash::Hash::hash(&organization_id, &mut hasher);
+                let org_id_hash = hasher.finish();
+                let shard = org_id_hash % shards;
+
+                // should be ok to unwrap since we MUST have at least one range defined
```

Comment:
This is reasonable but because it is so far apart from each other in the code I'm a bit uncomfortable about it.  One way would be make a custom type for the producers, then it is enforced in that type and relying on that invariant would be a lot nicer and keeps the code interacting with the invariant close together.  I guess this would be a newtype over the BTreeMap with a few small methods.

Reply:
That's a very good point. I will look into it. 
It might also simplify the code and removes some duplication.

Reply:
So, I moved the same functionality under the `Sharded` producer in store actore [d4a05c2](https://github.com/getsentry/relay/pull/1454/commits/d4a05c23f39c9238481d9338b0b6b7f73ebb7010). 
I do not know if it's a good idea to create a new type over `BTreeMap`, which can be a bit more complicated to understand.
But I might have to pick your brain on this , and how to implement better.